frontend/src/components/PaginatedTable.js

```diff
@@ -37,31 +37,27 @@ function PaginatedTable ({
   const filtersRef = useRef(filters);
   const sortByRef = useRef(sortBy);
   const [currentPage, setCurrentPage] = useState(1);
-  const nextPage = currentPage + 1;
-  const nextPageEndIndex = nextPage * pageSize;
 
   const startIndex = (currentPage - 1) * pageSize;
   const endIndex   = Math.min((currentPage) * pageSize, totalCount);
 
-  const isLastPage = endIndex >= totalCount;
-
   const dataSource =
     items.slice(startIndex, endIndex)
       .map(id => itemsByID[id]);
 
   const hasUnloadedItems = dataSource.some(d => d === undefined);
   const isCurrentPageUnloaded = ((endIndex - 1) > items.length) || hasUnloadedItems;
-  const doesNextPageContainUnloaded = !isLastPage && nextPageEndIndex > items.length && items.length < totalCount;
-  const shouldLoadNextChunk =
-    !loading && (isCurrentPageUnloaded || doesNextPageContainUnloaded);
+  const shouldLoadNextChunk = !loading && isCurrentPageUnloaded;
 
   if (shouldLoadNextChunk) {
     let offset
 
-    if (isCurrentPageUnloaded)
+    if (isCurrentPageUnloaded){
```

Comment:
I think that we can just get rid of the `isCurrentPageUnloaded` test here and simply compute the offset. We want the same offset if `hasUnloadedItems`  or '`isCurrentPageUnloaded` is true. If we are on page six and `hasUnloadedItems` is true, but we ask for the items on page 0, we will never load the page 6 items and go into an infinite loop.